src/core/probe/manager.rs

```diff
@@ -293,9 +297,36 @@ impl ProbeManager {
             match builders.contains_key(&probe.type_key()) {
                 false => {
                     let mut builder: Box<dyn ProbeBuilder> = match probe.r#type() {
-                        ProbeType::Kprobe(_) => Box::new(kprobe::KprobeBuilder::new()),
-                        ProbeType::Kretprobe(_) => Box::new(kretprobe::KretprobeBuilder::new()),
+                        ProbeType::Kprobe(_) => {
+                            register_filter_handler(
+                                "kprobe/probe",
```

Comment:
> Instead of using this generic `kprobe/probe` section type (needed if we'd use the automatic attach way, but we don't), we could rename it to `retis/kprobe` (and do the same for all `SEC`). This would reduce the risk of our handler running on non-retis probes.
> 
> I'm also wondering if we could use `bpf_program__name` in `fixup_filter_load_fn` to reduce again the risk (in addition to the `SEC` and magic number). That might be a little too much though.
> 

I'm not sure I follow, this whole thing should be local to the process, so unless we decide to load use non-retis probes, it should be safe.

> Also, I think we should also call `register_filter_handler` before this function, so it's called only once. If our probe sections are more specific, we can add an handler for all known types even if not actually used.

as it is, `register_filter_handler()` needs to be called more than once because the sections matter. Can you please elaborate "we should also call `register_filter_handler` before this function, so it's called only once"
This was already suggested to be moved here from the probe builders (from the logical point of view this part is fairly [lib]bpf dependent and felt naturally belonging there).

Reply:
the reasoning makes sense.
I moved it. I realized that some things remained in the builder didn't make much sense there, so I moved some stuff to the manager giving them all the dyn_filter hat.

Reply:
one more thing about "retis/" section naming.
Although makes sense to rename it in this PR, I think this could be considered as a different logical change.
Although it should be fairly safe, we have a lot on the plate with this PR, and doing it in a follow-up could make sense (at least we could test that change on its own). WDYT?